Relax type signature for tools.codegen.api.translate

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51590 Support at::cpu on non-structured kernels
* #51585 Split anonymous and namespaced definitions in RegisterDispatchKey
* #51583 Factor out structured generation into its own subclass.
* #51508 Split out RegisterDispatchKey to its own file
* #51500 Use Literal to model targets.
* #51499 Add support for generating faithful at::cpu signatures
* #51490 Add api.structured; switch structured kernels to use const Tensor& everywhere
* **#51477 Relax type signature for tools.codegen.api.translate**

Passing in a full binding is still OK, but if you have less
(e.g., an Expr/CType), that will do too. I'll need this for
some codegen patches I'm doing later.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26179560](https://our.internmc.facebook.com/intern/diff/D26179560)

ezyang committed:
Relax type signature for tools.codegen.api.translate

Passing in a full binding is still OK, but if you have less
(e.g., an Expr/CType), that will do too. I'll need this for
some codegen patches I'm doing later.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Relax type signature for tools.codegen.api.translate"


Passing in a full binding is still OK, but if you have less
(e.g., an Expr/CType), that will do too. I'll need this for
some codegen patches I'm doing later.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26179560](https://our.internmc.facebook.com/intern/diff/D26179560)

[ghstack-poisoned]

bhosmer commented on tools/codegen/api/translate.py:
```diff
@@ -44,14 +44,33 @@ class UnsatError(RuntimeError):
 # and you need to generate "exprs".
-#
-# TODO: Don't need full Binding for goals, CType will do
-# TODO: Don't need full Binding for bindings, list of Expr will do
```
I think it's a short comment header documenting the params - even just the signal in these TODOs would be helpful to someone who's unfamiliar with this code

ezyang committed:
Update on "Relax type signature for tools.codegen.api.translate"


Passing in a full binding is still OK, but if you have less
(e.g., an Expr/CType), that will do too. I'll need this for
some codegen patches I'm doing later.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D26179560](https://our.internmc.facebook.com/intern/diff/D26179560)

[ghstack-poisoned]
